### PR TITLE
feat: Add AWS Cloud scanning

### LINKED ISSUE #2476: AWS scanning
Add a new target to trivy that scans an AWS account. By connecting to the AWS API, discovering relevant resources, and scanning them with current scanners. For starts we will scan for misconfigurations only.
the UX flow should be similar to Kubernetes scanning: start by a broad scan, and show summary only. then user can drill into resources to detailed findings.

Things to consider:
- how to consistently specify an AWS resource (id/name/arn)
- local caching of resources. to allow a responsive experience while "navigating" within an account

### PR DESCRIPTION
**Pull request by liamg**
## Description
Adds the ability to scan live AWS cloud accounts, using the same rules as supplied by defsec for misconfiguration scanning.

The rules currently cover the [AWS CIS 1.2](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-standards-cis.html) benchmarks.

See the [docs](https://github.com/aquasecurity/trivy/blob/liamg-aws-poc/docs/docs/cloud/aws/scanning.md) for more information.

## Related issues
- Close #2476 

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [x] I've added usage information (if the PR introduces new options)
- [x] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -327,3 +380,5 @@ replace github.com/docker/docker => github.com/docker/docker v20.10.3-0.20220224
 // v1.2.0 is taken from github.com/open-policy-agent/opa v0.42.0
 // v1.2.0 incompatible with github.com/docker/docker v20.10.3-0.20220224222438-c78f6963a1c0+incompatible
 replace oras.land/oras-go => oras.land/oras-go v1.1.1
+
+replace github.com/elgohr/go-localstack => github.com/aquasecurity/go-localstack v0.0.0-20220706080605-1ec0e9b8753c
```
Comment (knqyf263) [inline go.mod:384]: Do you think our changes have a chance to get merged upstream?

File: pkg/commands/app.go [hunk 2]
```diff
@@ -787,6 +793,69 @@ func NewKubernetesCommand(globalFlags *flag.GlobalFlagGroup) *cobra.Command {
...
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if err := awsFlags.Bind(cmd); err != nil {
+				return xerrors.Errorf("flag bind error: %w", err)
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := awsFlags.Bind(cmd); err != nil {
+				return xerrors.Errorf("flag bind error: %w", err)
+			}
```
Comment (knqyf263) [inline pkg/commands/app.go:838]: Looks like we don't need to bind flags here as it is done in `PreRunE`.

File: pkg/commands/option/aws.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package option
```
Comment (knqyf263) [inline pkg/commands/option/aws.go:1]: This file seems to be stale.

File: pkg/fanal/analyzer/const.go [hunk 4]
```diff
@@ -100,6 +100,11 @@ const (
 	// =======
 	TypeRedHatContentManifestType Type = "redhat-content-manifest"
 	TypeRedHatDockerfileType      Type = "redhat-dockerfile"
+
+	// =====
+	// Cloud
+	// =====
+	TypeCloudAWS Type = "cloud-aws"
```
Comment (knqyf263) [inline pkg/fanal/analyzer/const.go:107]: I think it is not used.

**Review by knqyf263 — COMMENTED** [review]
LGTM. I left some small comments. Note I don't have an AWS account and couldn't test this PR against the live environment.

File: go.mod (hunk 1 above)
Comment (liamg) [reply go.mod:384]: I think @owenrumney already raised a PR there :+1:

**Comment by knqyf263** [comment]
```
$ ./trivy aws --region eu-central-1 --service ec2

Scan Overview for AWS Account XXXXXXXXXX
┌───────────────┬──────────────────────────────────────────────────┬────────────────┐
│               │                Misconfigurations                 │                │
│               ├──────────┬──────────────┬────────┬─────┬─────────┤                │
│ Service       │ Critical │     High     │ Medium │ Low │ Unknown │  Last Scanned  │
├───────────────┼──────────┼──────────────┼────────┼─────┼─────────┼────────────────┤
│ api-gateway   │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ athena        │        0 │            2 │      0 │   0 │       0 │ 22 minutes ago │
│ cloudfront    │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ cloudtrail    │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ cloudwatch    │        0 │            0 │      0 │   4 │       0 │ 22 minutes ago │
│ codebuild     │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ documentdb    │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ dynamodb      │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ ec2           │        5 │            1 │      0 │   0 │       0 │ 22 minutes ago │
│ ecr           │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ ecs           │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ efs           │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ eks           │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ elasticache   │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ elasticsearch │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ elb           │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ emr           │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ iam           │        0 │          507 │      0 │   8 │       0 │ 22 minutes ago │
│ kinesis       │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ kms           │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ lambda        │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ mq            │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ msk           │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ neptune       │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ rds           │        1 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ redshift      │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ s3            │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ sns           │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ sqs           │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ ssm           │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
│ workspaces    │        0 │            0 │      0 │   0 │       0 │ 22 minutes ago │
└───────────────┴──────────┴──────────────┴────────┴─────┴─────────┴────────────────┘

This scan report was loaded from cached results. If you'd like to run a fresh scan, use --update-cache.
```

Is this behavior intended? Seems like it scans all the resources even with `--update-cache`.

```
$ ./trivy aws --region eu-central-1 --service ec2 --update-cache
[1/31] Scanning api-gateway...
[2/31] Scanning athena...
[3/31] Scanning cloudfront...
[4/31] Scanning cloudtrail...
[5/31] Scanning cloudwatch...
[6/31] Scanning codebuild...
[7/31] Scanning documentdb...
[8/31] Scanning dynamodb...
[9/31] Scanning ec2...
[10/31] Scanning ecr...
[11/31] Scanning ecs...
[12/31] Scanning efs...
```

**Comment by knqyf263** [comment]
I confirmed https://github.com/aquasecurity/trivy/pull/2493/commits/633034d377a999832bafc7b1dc6a51d3c54691d5 worked.

**Review by knqyf263 — APPROVED** [review]

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**